sentry/backend.go

```diff
@@ -0,0 +1,226 @@
+// The Sentry package contains an implementation of a bridge between
+// the sentry-go package and glog which allows for glog errors of
+// level ERROR to be tracked as errors in Sentry. This includes
+// custom code that interfaces with the xerrors package (as well
+// as the Yext fork named yerrors) in order to send stack trace
+// data for the glog invocation, error object construction,
+// and any masked error calls to Sentry.
+package sentry
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/yext/glog"
+	"github.com/yext/glog-contrib/stacktrace"
+)
+
+// The maximum number of wrapped errors processed.
+const maxErrorDepth = 10
+
+var (
+	sentryDebug = flag.Bool("sentryDebug", false,
+		"enable debug mode in Sentry clients")
+	sentryFingerprinting = flag.Bool("sentryFingerprinting", false,
+		"enable server-side issue fingerprinting. If set, duplicate issues will only be tracked if they have equivalent filenames and line numbers")
+
+	hostname string
+)
+
+func init() {
+	hostname, _ = os.Hostname()
+	if short := strings.Index(hostname, "."); short != -1 {
+		hostname = hostname[:short]
+	}
+}
+
+// CaptureErrors is the entrypoint for tracking Sentry exceptions via glog.
+// Given Sentry DSNs and client options (DSN should not be specified in opts),
+// constructs individual Sentry Client's for each DSN. The glog.Event channel
+// should be provided by running glog.RegisterBackend(). For example:
+//  sentry.CaptureErrors(
+//  	"projectName",
+//  	[]string{"https://primaryDsn", "https://optionalSecondaryDsn", ...},
+//		sentrygo.ClientOptions{
+//			Release: "release",
+//			Environment: "prod",
+//		},
+//		glog.RegisterBackend())
+//
+// When an event is received via glog at the ERROR severity,
+// the first provided DSN will be used, unless a sentry.AltDsn is
+// tagged on the glog event, in which case the specified client
+// for that DSN will be used:
+//   glog.Error("error for secondary DSN", sentry.AltDsn("https://optionalSecondaryDsn"))
+func CaptureErrors(project string, dsns []string, opts sentry.ClientOptions, comm <-chan glog.Event) {
+	// If no DSNs specified, panic (we can't invoke glog)
+	if len(dsns) == 0 {
```

Comment:
It will fall back to the noop DSN if there's a single DSN specified with empty string. I guess it could be reasonable to have it do the same behavior if there are none? I copied the existing raven behavior for this: https://github.com/yext/glog-contrib/blob/master/raven/backend.go#L58

Reply:
Hmm, in that case, don't worry about it.